document-store/src/main/java/org/hypertrace/core/documentstore/expression/type/FilterTypeExpression.java

```diff
@@ -1,11 +1,11 @@
 package org.hypertrace.core.documentstore.expression.type;
 
-import org.hypertrace.core.documentstore.parser.FilteringExpressionVisitor;
+import org.hypertrace.core.documentstore.parser.FilterTypeExpressionVisitor;
 
 /**
  * An interface to represent that the expression can be used in either the WHERE clause or the
  * HAVING clause of the query.
  */
-public interface FilteringExpression {
-  <T> T visit(final FilteringExpressionVisitor visitor);
+public interface FilterTypeExpression {
```

Comment:
`FilterTypeExpression` seems lengthy and not easy to read. Can it just be `FilterExpression`?

Reply:
I initially thought of naming them `FilterExpression`, but there seemed to be a conflict in the implementation classes name and interface name. 

Basically I wanted to differentiate the implementation of these interfaces from the interface name itself, these interfaces are a logical grouping of expressions and any implementation is sort of belonging to that class of expression. 

For example AggregateExpression implements SelectTypeExpression, SortTypeExpression, so it belongs to two classes of expression. If the interface itself was called SelectExpression, its confusing to differentiate the implementation from the interface

Reply:
Yeah, I see your point. This naming suggestion came from [this comment](https://github.com/hypertrace/document-store/pull/58#discussion_r745136179). 

Now, I'm thinking we could probably go with `FilterableExpression`, `SelectableExpression`, `SortableExpression` and `GroupableExpression` so that we get both advantages via.

1. The `Expression` suffix indicates that it is an expression and not "something" which can be "filtered"
2. The `able` suffix indicates that this is an interface (and not a concrete implementation).

Reply:
The two points you listed is the precise to refactor this name. 
I am instead of `able` using `type` to distinguish interface from implementation. Given the package is also named `type` I felt it to be suitable.

Reply:
Agree with you on the fact that, whether it is the `type` suffix or the `able` suffix, it is going to add the same number of characters to the class name. But, with the `type` suffix, we have 3 words in the class name: `Filter`, `Type`, `Expression`. However, with the `able` suffix, we have 2 words in the class name: `Filterable`, `Expression`, which I feel is easier to read and discuss.